components/controls/ClipboardBase.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import * as Clipboard from 'clipboard';
+
+interface Props {
+  className?: string;
+  copyValue: string;
+}
+
+type ShowTooltipFunction = (() => never) | VoidFunction;
+
+export default class ClipboardBase extends React.PureComponent<Props> {
```

Comment:
I guess it's ok for now but I'm not a huge fan of having this component exported out of sonar-ui-common, since it's not usable on it's own...
Maybe a solution to this could be to do one `clipboard.tsx` file that would contain this base component (not exported) and the two other components that depend on it and would be exported.